### PR TITLE
Update Delivery: Done!

### PR DESCRIPTION
**Pull request by ertolv**
Update delivery in web works.

### CODE REVIEW

File: Web/utkeyrslukerfi-web/src/components/Delivery/index.jsx [hunk 1]
```diff
@@ -1,34 +1,63 @@
...
 
-  if (Object.entries(deliveryObj).length === 0) {
-    dispatch(getDelivery(token, id))
-  }
+
+  useEffect(() => {
+    dispatch(setDelivery(delivery))
+  }, [])
+
+  // if (Object.entries(delivery).length === 0) {
```
Comment (MaggaSK97) [inline Web/utkeyrslukerfi-web/src/components/Delivery/index.jsx:37]: Fjarlægja?

File: Web/utkeyrslukerfi-web/src/actions/deliveryActions.jsx [hunk 2]
```diff
@@ -15,9 +16,28 @@ const getDeliverySuccess = (delivery) => ({
...
+const setDelivery = (delivery) => ({
   type: SET_DELIVERY,
-  payload: obj
+  payload: delivery
+})
+
+const updateDelivery = (token, id, delivery) => async (dispatch) => {
+  try {
+    const res = await deliveryService.updateDelivery(token, id, delivery)
+    if (res?.status === 401) { toastr.error('401 error') }
```
Comment (MaggaSK97) [inline Web/utkeyrslukerfi-web/src/actions/deliveryActions.jsx:27]: Skrifa meira notendavænni villuskilaboð

File: Web/utkeyrslukerfi-web/src/components/DeliveryAddressModal/index.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react'
```
Comment (MaggaSK97) [inline Web/utkeyrslukerfi-web/src/components/DeliveryAddressModal/index.jsx:1]: Hafa eitt AddressModal component í stað Dalivery og Pickup

File: Web/utkeyrslukerfi-web/src/components/Delivery/index.jsx (hunk 1 above)
Comment (ertolv) [reply Web/utkeyrslukerfi-web/src/components/Delivery/index.jsx:37]: done

File: Web/utkeyrslukerfi-web/src/components/DeliveryAddressModal/index.jsx (hunk 3 above)
Comment (ertolv) [reply Web/utkeyrslukerfi-web/src/components/DeliveryAddressModal/index.jsx:1]: done!

File: Web/utkeyrslukerfi-web/src/actions/deliveryActions.jsx (hunk 2 above)
Comment (ertolv) [reply Web/utkeyrslukerfi-web/src/actions/deliveryActions.jsx:27]: done!

File: Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Models/InputModels/AddressInputModel.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace Utkeyrslukerfi.API.Models.InputModels
+{
+    public class AddressInputModel
+    {
+        [Required(ErrorMessage = "Street name is required")]
+        public string StreetName { get; set; }
+        [Required(ErrorMessage = "Home number is required")]
```
Comment (MaggaSK97) [inline Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Models/InputModels/AddressInputModel.cs:9]: house number

File: Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Repositories/Implementations/DeliveryRepository.cs [hunk 5]
```diff
@@ -79,22 +67,32 @@ public DeliveryDTO CreateDelivery(DeliveryInputModel delivery)
             // Get vehicle
             var vehicle = _dbContext.Vehicles.FirstOrDefault(v => v.ID == delivery.VehicleID);
             if (vehicle == null) { throw new NotFoundException("Vehicle not registered."); }
+            System.Console.WriteLine("Street name: ", delivery.PickupAddressStreetName);
```
Comment (MaggaSK97) [inline Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Repositories/Implementations/DeliveryRepository.cs:70]: remove?

File: Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Models/InputModels/AddressInputModel.cs (hunk 4 above)
Comment (ertolv) [reply Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Models/InputModels/AddressInputModel.cs:9]: resolved

File: Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Repositories/Implementations/DeliveryRepository.cs (hunk 5 above)
Comment (ertolv) [reply Backend/Utkeyrslukerfi.API/Utkeyrslukerfi.API.Repositories/Implementations/DeliveryRepository.cs:70]: sorry, debugging

**Pull request merged**